be/src/storage/storage_engine.cpp

```diff
@@ -456,12 +457,11 @@ void StorageEngine::stop() {
     {
         std::lock_guard<std::mutex> l(_store_lock);
         for (auto& store_pair : _store_map) {
+            // store_pair.second will be delete later
             store_pair.second->stop_bg_worker();
-            delete store_pair.second;
```

Comment:
Why move this delete in the end?

Reply:
store_pair.second will be used in compaction thread. so we need delete it after all of the background thread exit